tests/main/cifs-home/task.yaml

```diff
@@ -0,0 +1,131 @@
+            ;;
+        esac
+        ;;
+    esac
+
+    # Install samba.
+    case "$SPREAD_SYSTEM" in
+      ubuntu-2[24].04-*)
+        DEB_FRONTEND=noninteractive apt install -y samba
+        tests.cleanup defer apt remove --purge -y samba
+        ;;
+    esac
+
+    # Create the var-home share.
+    tests.backup prepare /etc/samba
+    tests.cleanup defer tests.backup restore /etc/samba
+    cat <<-__SMB__ >>'/etc/samba/smb.conf'
+    [var-home]
+       comment = Roaming home directories
+       path = /var/home
+       browseable = no
+       read only = no
+       create mask = 0755
+       directory mask = 0755
+       smb3 unix extensions = yes
+       ea support = yes
+    __SMB__
+    systemctl restart smbd.service
+
+    # Create a user called test-remote
+    adduser --uid 54321 --quiet --disabled-password --gecos '' test-remote
+    tests.cleanup defer deluser test-remote
+
+    # Allow user test-remote to authenticate to samba, with the password 'secret'.
+    printf 'secret\nsecret\n' | smbpasswd -a test-remote
+    tests.cleanup defer smbpasswd -x test-remote
+
+    # Later on, restart snapd and ensure that nfs/cifs workaround is gone.
+    # This cleanup handler is registered before we mount the cifs file system.
+    if [ "$(snap debug confinement)" = strict ]; then
+        tests.cleanup defer test ! -e /var/lib/snapd/apparmor/snap-confine/nfs-support
+    fi
+    tests.cleanup defer systemctl start snapd.service
+    tests.cleanup defer systemctl reset-failed snapd.service snapd.socket
+    tests.cleanup defer systemctl stop snapd.service snapd.socket
+
+    # Move the actual home directory of the user test-remote to /var/home and mount /home/test-remote from samba.
+    if [ ! -d /var/home ]; then
+      mkdir /var/home
+      tests.cleanup defer rmdir /var/home
+    fi
+    mkdir /var/home/test-remote
+    chown test-remote.test-remote /var/home/test-remote
+    tests.cleanup defer rm -rf /var/home/test-remote
+    # TODO: it would be nice to successfully pass ",posix" option and get things to mount.
+    mount -t cifs //127.0.0.1/var-home/test-remote /home/test-remote -o vers=3.1.1,nomapposix,mfsymlinks,username=test-remote,password=secret,uid=54321,gid=54321
+    tests.cleanup defer umount /home/test-remote
+
+    # Ensure the session is working.
+    tests.session -u test-remote prepare
+    tests.cleanup defer tests.session -u test-remote restore
+
+    # Install local copy of test-snapd-sh which plugs in the home interface but not the network interface.
+    "$TESTSTOOLS"/snaps-state install-local test-snapd-sh
+
+    # Check that snapd works around CIFS apparmor interaction. Note that the
+    # name of the file is "nfs-support" as we are not trying to change too much
+    # of the extension interface visible to the system.
+    systemctl stop snapd.service snapd.socket
+    systemctl reset-failed snapd.service snapd.socket
+    systemctl start snapd.service
```

Comment:
why not simply restart?

Reply:
This part came from nfs-support test. I don't recall, perhaps it's from reset-failed making restart a no-op? Dunno.

Reply:
I've simplified this now.